pkg/k8s/client.go

```diff
@@ -35,82 +37,94 @@ const (
 
 // KubeInfo holds the data required to talk to a cluster
 type KubeInfo struct {
-	Kubeconfig   string
-	context      string
-	namespace    string
-	clientConfig clientcmd.ClientConfig
+	flags       *pflag.FlagSet
+	configFlags *genericclioptions.ConfigFlags
+	config      *rest.Config
+	namespace   string
 }
 
 // NewKubeInfo returns a useable KubeInfo, handling optional
 // kubeconfig, context, and namespace.
 func NewKubeInfo(configfile, context, namespace string) *KubeInfo {
-	// Find the correct kube config file
-	configfilesearch := clientcmd.NewDefaultClientConfigLoadingRules()
-	if len(configfile) != 0 {
-		configfilesearch.ExplicitPath = configfile
-	}
+	flags := pflag.NewFlagSet("KubeInfo", pflag.PanicOnError)
+	result := NewKubeInfoFromFlags(flags)
 
-	// Possibly override context and namespace
-	overrides := &clientcmd.ConfigOverrides{}
-	if len(context) != 0 {
-		overrides.CurrentContext = context
+	var args []string
+	if configfile != "" {
+		args = append(args, "--kubeconfig", configfile)
 	}
-	if len(namespace) != 0 {
-		overrides.Context.Namespace = namespace
+	if context != "" {
+		args = append(args, "--context", context)
+	}
+	if namespace != "" {
+		args = append(args, "--namespace", namespace)
 	}
 
-	// Construct the config
-	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(configfilesearch, overrides)
-
-	res := KubeInfo{
-		configfile,
-		context,
-		namespace,
-		kubeconfig,
+	err := flags.Parse(args)
+	if err != nil {
+		panic(err)
```

Comment:
I wrote 15 days ago:
> ~~I'd rather return the error (even though adding an error to the signature would be a breaking change to callers).~~
>
> Can we add a comment that since `args` is constructed by us, and there should never be an error, it's OK to `panic`?

Reply:
Done